muspinsim/__main__.py

```diff
@@ -24,29 +24,73 @@
 LOGFORMAT = "[%(levelname)s] [%(threadName)s] [%(asctime)s] %(message)s"
 
 
+def check_dir_path(string):
```

Comment:
This function name could be clearer that it will create directories if they don't exist. Functions with names like `check_x` often act only as validation and tend not to manipulate things, so this is a slightly confusing break from (admittedly not super well-established) convention. I'd suggest something like `ensure_dir_path_exists` instead.